src/util/s3select.js

```diff
@@ -74,6 +108,8 @@ class S3SelectStream extends Transform {
         const buffer = Buffer.alloc(S3SelectStream.prelude_length_bytes);
         let index = 0;
 
+        this.stats.Stats.BytesReturned += select_result.select.length;
```

Comment:
Maybe you don't need the `Stats`? (change it where you defined it).
Then it will be easier to read/write  `this.Stats.BytesReturned` instead of `this.stats.Stats.BytesReturned`

Reply:
The "Stats" object has to be created sooner or later to be encoded into xml and sent as part of finishing the request (in send_stats()).
Creating it "sooner" saves me creating it explicitly later and copying "stats" into it, though it does add this extra ".stats" in some lines.
I prefer it this way, but it is a not-critical matter of style.

Reply:
I don't understand why do you need `stats`?
Why not just:
```
        this.Stats = {
                BytesProcessed: 0,
                BytesScanned: 0,
                BytesReturned: 0
        };
```

Reply:
The issue is what object is sent to xml_utils.encode_xml().
This object has to conform to AWS structure (as defined in their specs).
Your object does not.
It's possible to go with your object, and manully create the conforming object in send_stats(), but I'm not sure it's significantly better.